Live chat/add the ability for profanity to be toggled in the users settings

aflekkas opened the pull request:
This addresses issue #920. This provides the final scaffolding needed for @pmdepaulo to implement her profanity filtering features.

aflekkas committed:
Added profanity toggle and url for profanity toggle (now we need to do it visually too)

aflekkas committed:
Added a button to toggle the profanity filter in the user profile detail view.

pmdepaulo reviewed (approved):
Thank you, this is super helpful! Your use of documentation is great here. Just as a heads up, when I branch off of this to continue working on it, I might explore being able to toggle profanity on/off for specific live chats as opposed to having one toggle for the entire user, but this approach works as well.

aflekkas commented:
> Thank you, this is super helpful! Your use of documentation is great here. Just as a heads up, when I branch off of this to continue working on it, I might explore being able to toggle profanity on/off for specific live chats as opposed to having one toggle for the entire user, but this approach works as well.

Perhaps we can have both in a way? A master level toggle but nullable options per chat.

Allowing the setting to either inherit from the user or if it is set override the users setting.

Thanks for your feedback!

abargher reviewed (changes requested):
Toggle mechanism looks good, but there are some merge conflicts from dev. Take a look at those and you're all set!

quincyleung commented:
@alexandros-lekkas Going to join in on this PR to help with some UI improvements and also fix merge conflicts!

aflekkas commented:
🙏 @quincyleung ty

could you send some SS of the progress?

Commit:
added profanity to models

Commit:
added urls for profanity toggle

Commit:
added views for profanity toggle

Commit:
redid views

Commit:
added button to toggle

Commit:
added profanity template tags

Commit:
added conditional for profanity

Commit:
edited views

Commit:
functional toggle button for profanity

Commit:
edited views toggle profanity

nathanL534 committed:
Merge branch 'dev' into live-chat/add-public-private-chat-toggle

quincyleung commented:
Currently running into this issue 
<img width="1409" alt="Screenshot 2025-05-23 at 2 59 49 PM" src="https://github.com/user-attachments/assets/763c3c84-652d-4cfe-9fca-beeab74735b3" />

quincyleung commented:
Will fix and make UI improvements!

quincyleung commented:
https://github.com/user-attachments/assets/c8500567-9fb6-4af6-9781-bf8645da99f3

Switched to toggle for more intuitive UI!

quincyleung committed:
Improve toggle profanity UI

pmdepaulo reviewed (approved):
This looks good to me!

quincyleung committed:
Merge branch 'dev' into live-chat/add-public-private-chat-toggle

abargher committed:
Merge branch 'dev' into live-chat/add-public-private-chat-toggle

abargher reviewed (changes requested):
This is a wide-reaching feature, so I'm not surprised there's some cross-team merge conflicts. Please take a careful look at these.

Commit:
fixed merge

quincyleung commented:
Please merge #1414 (backend for leaving chats) before merging this one!

Commit:
Merge branch 'live-chat/improve-main-chat-page-ui' into live-chat/add-public-private-chat-toggle

umerp committed:
[Tournaments] Add Live Character Counters and Character Limit to Tournament Creation (#1373)

* Add knowledge base migration before merging dev

* I have added  Live Character Counters and Character Limit to make sure games can be started with ease resloves issue #1366

* Remove unnecessary migration file from knowledge_base

* Trigger GitHub conflict resolution refresh

* fixed issues

Abebes01 committed:
[games] Implement weighted recommendation algorithm (#1121)

* [games] Implement weighted recommendation algorithm

Added new function that recommends similar games based on
multiple weighted factors including categories, mechanics,
and designers.

Closes: #874

* trigger comment

---------

Co-authored-by: ekabaeva <122409534+ekabaeva@users.noreply.github.com>
Co-authored-by: Sam Xie <72187977+MuyanXie@users.noreply.github.com>

quincyleung committed:
Merged dev into branch and resolved conflict. Ready to re-add toggle UI

quincyleung committed:
Fix profanity toggle logic and conflicts

quincyleung commented:
https://github.com/user-attachments/assets/7cd10885-6e01-4dc6-9e78-3e0a870bdf51

Commit:
fixed merge conflcits

Commit:
Merge remote-tracking branch 'origin/dev' into live-chat/add-conditional-rendering-of-profanity-filters

Commit:
[live chat] conditional profanity for individual chats

nathanL534 committed:
Merge branch 'dev' into live-chat/add-conditional-rendering-of-profanity-filters

Commit:
[live chat] cleande up code

Commit:
Merge branch 'live-chat/add-conditional-rendering-of-profanity-filters' of github.com:uchicago-cs/chigame into live-chat/add-conditional-rendering-of-profanity-filters

nathanL534 committed:
Merge branch 'dev' into live-chat/add-conditional-rendering-of-profanity-filters

quincyleung committed:
Fix merge conflicts

quincyleung committed:
Fix linting errors

quincyleung committed:
Add docstrings for profanity toggling

abargher reviewed (commented):
Looks great! Approved, and got merged along with #1393.

abargher commented:
Issue Score: Excellent

Comments:
Great work!

<!-- {"score": "excellent", "minor": false} -->
